src/MaxMind/Db/Reader.php

```diff
@@ -237,8 +238,10 @@ private function findMetadataStart($filename)
         $fileSize = $fstat['size'];
         $marker = self::$METADATA_START_MARKER;
         $markerLength = self::$METADATA_START_MARKER_LENGTH;
+        $maxMetadataSize = self::$METADATA_MAX_SIZE;
+        // $maxMetadataSize = $fileSize - $markerLength;
 
-        for ($i = 0; $i < $fileSize - $markerLength + 1; $i++) {
+        for ($i = 0; $i < min($maxMetadataSize, $fileSize - $markerLength) + 1; $i++) {
```

Comment:
Also, the `+ 1` should probably be moved inside the `min` as `$fileSize-$maxMetadataSize` is where the metadata would start for the max size, not `+ 1`.